eval/eval.go

```diff
@@ -1218,6 +1227,31 @@ func asObjectOrNil(v any) map[string]esc.Value {
 	return cast
 }
 
+// evaluateBuiltinConcat evaluates a call to the fn::concat builtin.
+func (e *evalContext) evaluateBuiltinConcat(x *expr, repr *concatExpr) *value {
+	v := &value{def: x, schema: x.schema}
+
+	arrays, ok := e.evaluateTypedExpr(repr.arrays, schema.Array().Items(schema.Array().Items(schema.Always())).Schema())
+	if !ok {
+		v.unknown = true
+		return v
+	}
+
+	v.combine(arrays)
+	if !v.unknown {
+		var result []*value
+		for _, arrayVal := range arrays.repr.([]*value) {
+			if arrayVal.repr != nil {
+				if arr, isArray := arrayVal.repr.([]*value); isArray {
+					result = append(result, arr...)
+				}
+			}
```

Comment:
I _think_ you should be able to just do `append` here w/o the nil + type checks:

```suggestion
			result = append(result, arrayVal.repr.([]*value)...)
```

The type checker should have issued errors otherwise.